test/CompressedBlockPrefilteringTest.cpp

```diff
@@ -0,0 +1,887 @@
+        referenceIdChild1, referenceIdChild2);
+    if constexpr (std::is_same_v<ResT, std::string>) {
+      testThrowError(std::move(expression), evaluationColumn, input, expected);
+    } else {
+      static_assert(std::is_same_v<ResT, std::vector<BlockMetadata>>);
+      EXPECT_EQ(expression->evaluate(input, evaluationColumn), expected);
+    }
+  }
+};
+
+//______________________________________________________________________________
+template <typename ResT>
+struct TestNotExpression {
+  template <typename RelExpr1, typename LogExpr = std::nullptr_t,
+            typename RelExpr2 = std::nullptr_t>
+  void test(size_t evaluationColumn, const std::vector<BlockMetadata>& input,
+            const ResT& expected, const ValueId& referenceId1,
+            std::optional<ValueId> optId2 = std::nullopt)
+      requires check_is_relational_v<RelExpr1> {
+    auto expression =
+        makeNotExpression<RelExpr1, LogExpr, RelExpr2>(referenceId1, optId2);
+    if constexpr (std::is_same_v<ResT, std::string>) {
+      testThrowError(std::move(expression), evaluationColumn, input, expected);
+    } else {
+      static_assert(std::is_same_v<ResT, std::vector<BlockMetadata>>);
+      EXPECT_EQ(expression->evaluate(input, evaluationColumn), expected);
+    }
+  }
+};
+
+//______________________________________________________________________________
+TEST(BlockMetadata, testBlockFormatForDebugging) {
+  MetadataBlocks blocks{};
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: V:10 D:33.000000 I:0\n(last) Triple: "
+      "V:10 D:33.000000 I:0\nnum. rows: 0.\n",
+      (std::stringstream() << blocks.b1).str());
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: V:10 D:33.000000 I:-4\n(last) Triple: "
+      "V:10 D:33.000000 D:2.000000\nnum. rows: 0.\n",
+      (std::stringstream() << blocks.b7).str());
+  EXPECT_EQ(
+      "#BlockMetadata\n(first) Triple: V:10 D:33.000000 V:14\n(last) Triple: "
+      "V:10 D:33.000000 V:17\nnum. rows: 0.\n",
+      (std::stringstream() << blocks.b17).str());
+}
+
+//______________________________________________________________________________
+// Test LessThanExpression
+TEST(RelationalExpression, testLessThanExpressions) {
+  MetadataBlocks blocks{};
+  TestRelationalExpression<LessThanExpression, std::vector<BlockMetadata>>
+      testExpression;
+  testExpression(
+      2, IntId(5), blocks.blocks,
+      {blocks.b1, blocks.b2, blocks.b5, blocks.b6, blocks.b7, blocks.b8,
+       blocks.b9, blocks.b10, blocks.b11, blocks.b12, blocks.b13, blocks.b14});
+
+  testExpression(2, IntId(-12), blocks.blocks, {blocks.b7, blocks.b14});
+  testExpression(2, IntId(0), blocks.blocks,
+                 {blocks.b5, blocks.b6, blocks.b7, blocks.b11, blocks.b12,
+                  blocks.b13, blocks.b14});
+  testExpression(2, IntId(100), blocks.blocks,
+                 {blocks.b1, blocks.b2, blocks.b3, blocks.b4, blocks.b5,
+                  blocks.b6, blocks.b7, blocks.b8, blocks.b9, blocks.b10,
```

Comment:
And if you write this test inside a test fixture (TEST_F in Gtest, just google it),which has the blocks b1 - b... directly as members you can always write `b1` instead of `blocks.b1` which also increases the readability.